### ISSUE TITLE
Arithmetic shift conflicts with heredoc end markers

### ISSUE THREAD

**Issue by Mark-E-Hamilton** [SimonKagstrom/kcov#66]
Does kcov have some sort of upper limit on the shell script size? We have some rather large scripts that I've been trying to check, and the report page shows the following:
````
3044   216       declare -r NIGHTLY_CODE_PROJECT_MASK=$(( 1 << 0 ))
3045                declare -r NIGHTLY_TESTS_PROJECT_MASK=$(( 1 << 1 ))
3046                declare -r NIGHTLY_DOCS_PROJECT_MASK=$(( 1 << 2 ))
````
Line 3044 is green, and many lines before it are either red or green. However, all lines after line 3044 are white. In particular, line 3045 is white, and if it executed 3044 there's no obvious way for it to not then execute line 3045. FWIW, the 'Instrumented lines' field show 1213 lines, with 607 executed.

**Comment by SimonKagstrom** [comment]
No, there's no size limit - kcov allocates lines dynamically so apart until memory runs out, it should handle any size. I believe this is a problem with kcov bash parsing (again!). This snippet

```sh
echo Before
declare -r NIGHTLY_CODE_PROJECT_MASK=$(( 1 << 0 ))
declare -r NIGHTLY_TESTS_PROJECT_MASK=$(( 1 << 1 ))
declare -r NIGHTLY_DOCS_PROJECT_MASK=$(( 1 << 2 ))
echo After
```

also gives the same error. I think it's the same root-cause as Issue #50: The "<< 0 ))" stuff is parsed as a here-document, so kcov will look for a terminating "0 ))" after that line.

**Comment by SimonKagstrom** [comment]
Fixed in a78360f.

**Comment by Mark-E-Hamilton** [comment]
I noticed that another large script of ours didn't exhibit this issue, and was going to track it down this morning, but now I don't need to. Thanks.
